nakadi-java-client/src/main/java/nakadi/CompressionSupport.java

```diff
@@ -0,0 +1,40 @@
+package nakadi;
+
+import java.io.IOException;
+/**
+ * Provides compression support for the client.
+ */
+public interface CompressionSupport {
+
+  /**
+   * Compress the supplied bytes
+   *
+   * @param bytes the target to compress
+   * @return the compressed bytes
+   */
+  public byte[] compress(byte[] bytes);
+
+  /**
+   * Compress the supplied String
+   *
+   * @param json the target to compress
+   * @return the compressed bytes
+   */
+  byte[] compress(String json);
+
+  /**
+   * Decompress the supplied array.
+   *
+   * @param compressed the target to de compress
+   * @return the decompressed String
+   */
+  String decompress(byte[] compressed);
```

Comment:
Seems that this method is required for tests, not for the actual client, and may be removed from public interface.

Reply:
Thank you. I'll mark it ahead of the release.